coins/monero/src/wallet/seed/polyseed.rs

```diff
@@ -375,9 +387,12 @@ impl Polyseed {
     let features =
       u8::try_from(extra >> DATE_BITS).expect("couldn't convert extra >> DATE_BITS to u8");
 
-    let res = Polyseed::from(lang, features, birthday_decode(birthday), entropy);
+    let res = from_internal(lang, features, birthday, entropy);
     if let Ok(res) = res.as_ref() {
-      debug_assert_eq!(res.checksum, checksum);
+      if res.checksum != checksum {
+        // This should never trigger
```

Comment:
If this should never trigger, the debug_assert is valid.

Reply:
Ok, why should it be a debug assert and not a normal assert? I see no reason the function should return a normal response in a non-debug build if this statement is true. The seed would be invalid.

Reply:
I guess debug assert in one line, then return the invalid seed error in the next could make sense.

Reply:
Because it should be impossible, and is accordingly redundant, so it should be optimized out on release.